v9 (pnpm): new Expo Prebuild project missing expo-modules-core

### Describe the bug

Project initialization:
```console
➜ npx ignite-cli@latest new PizzaApp

✅ What bundle identifier? · com.pizzaapp
✅ Where do you want to start your project? · C:\bugreport\PizzaApp
✅ Choose a workflow: · Expo Prebuild
✅ Do you want to initialize a git repository? (Y/n) · Yes
✅ Remove demo code? We recommend leaving it in if it's your first time using Ignite (y/N) · No
✅ Which package manager do you want to use? · pnpm
✅ Do you want to install dependencies? (Y/n) · Yes
✅ ❗EXPERIMENTAL❗Would you like to enable the New Architecture? (y/N) · No
```

Running `pnpm run android` results in the following:
```console
➜ pnpm run android

> pizza-app@0.0.1 android C:\bugreport\PizzaApp
> npx expo run:android

› Building app...
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\bugreport\PizzaApp\node_modules\expo\android\build.gradle' line: 39

* What went wrong:
A problem occurred evaluating project ':expo'.
> Project with path ':expo-modules-core' could not be found in project ':expo'.
```

### Ignite version

9.0.0

### Additional info

```console
➜ npx ignite-cli@latest doctor
System
  platform           win32
  arch               x64
  cpu                24 cores     13th Gen Intel(R) Core(TM) i7-13700K
  directory          PizzaApp     C:\bugreport\PizzaApp

JavaScript
  node               18.16.0                                                                             C:\Program Files\nodejs\node.EXE
  npm                9.5.1                                                                               C:\Program Files\nodejs\npm.CMD
  yarn               -                                                                                   not installed
  pnpm               8.10.0                                                                              C:\Users\lucas\AppData\Local\pnpm\pnpm.EXE
                     <no pnpm global package info available>
  bun                -                                                                                   not installed
  expo               49.0.18 -> .\node_modules\.pnpm\expo@49.0.18_@babel+core@7.23.3\node_modules\expo   managed

Ignite
  ignite-cli         9.0.0        C:\Users\lucas\AppData\Local\npm-cache\_npx\bec2839915ba8fc3\node_modules\.bin\ignite.CMD
  ignite src         build        C:\Users\lucas\AppData\Local\npm-cache\_npx\bec2839915ba8fc3\node_modules\ignite-cli\build

Android
  java               17.0.6       C:\Users\lucas\.jdks\temurin-17.0.6\bin\java.EXE
  android home       -            C:\Users\lucas\AppData\Local\Android\Sdk

Tools
  git                git version 2.40.0.windows.1   C:\Program Files\Git\cmd\git.EXE
```

--- a/src/commands/new.ts
+++ b/src/commands/new.ts
@@ -575,6 +575,14 @@ module.exports = {
       // #region Run Packager Install
       // pnpm/yarn/npm/bun install it
 
+      // fix .npmrc if using pnpm
+      if (packagerName === "pnpm") {
+        // append `node-linker=hoisted` to .npmrc
+        const npmrcPath = path(targetPath, ".npmrc")
+        const npmrcContents = read(npmrcPath)
+        write(npmrcPath, `${npmrcContents}${EOL}node-linker=hoisted${EOL}`)
+      }
+
       // check if there is a dependency cache using a hash of the package.json
       const boilerplatePackageJsonHash = cache.hash(read(path(boilerplatePath, "package.json")))
       const cachePath = path(cache.rootdir(), boilerplatePackageJsonHash, packagerName)

